pkg/errors/armerrors.go

```diff
@@ -15,12 +15,81 @@ limitations under the License.
 package errors
 
 import (
+	"bytes"
+	"encoding/json"
 	"errors"
+	"fmt"
+	"io"
 	"strings"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 )
 
+type CreationResponseError struct {
+	*azcore.ResponseError
+}
+
+// TODO - this is a very WIP dummy implementation, trying to gather thoughts first before I make it more robust
+func (e *CreationResponseError) Error() string {
```

Comment:
Adjust this as needed - I'm looking through example errors that are returned to understand what sort of information we should focus on, this is the first "re-arrangement" based on the discussion in https://github.com/Azure/karpenter-provider-azure/pull/874#discussion_r2090038218

Reply:
Great to see the direction!

Not sure about `Creation...` in the name - we could probably use this in other contexts as well.

Potential guiding principles: intended for human consumption, targeting single line (should avoid new lines), is likely to be truncated and should therefore be concise (if possible) and have the most important info first.